scripts/interactions.js

```diff
@@ -61,14 +61,25 @@ function keypress(e){
 	var unicode=e.keyCode? e.keyCode : e.charCode;
 	console.log(unicode);
 	if(unicode === 46){ // Delete
-		// REMOVE SELECTION (from local graph only, this is not a deletion in remote database)
-		//for(node in graph.selection)
-		for(var i=0; i< graph.selection.length;i++)
+		if(graph.selection.length > 0)
 		{
-			var node = graph.selection[i];
-			console.log(node);
-			graph.removeNode(node);
+			// REMOVE SELECTION (from local graph only, this is not a deletion in remote database)
+			//for(node in graph.selection)
+			for(var i=0; i< graph.selection.length;i++)
+			{
+				var id;
+				(graph.selection[i]._id) ? id = graph.selection[i]._id : id = graph.selection[i].data._id;
```

Comment:
in selection, we should only have common object types, not springy types, graph-common and interactions will depend on springy which we would not prefer..